internal/cmd/argus/scrape-config/update/update.go

```diff
@@ -0,0 +1,130 @@
+	"github.com/stackitcloud/stackit-cli/internal/pkg/errors"
+	"github.com/stackitcloud/stackit-cli/internal/pkg/examples"
+	"github.com/stackitcloud/stackit-cli/internal/pkg/flags"
+	"github.com/stackitcloud/stackit-cli/internal/pkg/globalflags"
+	"github.com/stackitcloud/stackit-cli/internal/pkg/print"
+	"github.com/stackitcloud/stackit-cli/internal/pkg/services/argus/client"
+
+	"github.com/spf13/cobra"
+	"github.com/stackitcloud/stackit-sdk-go/services/argus"
+)
+
+const (
+	jobNameArg = "JOB_NAME"
+
+	instanceIdFlag = "instance-id"
+	payloadFlag    = "payload"
+)
+
+type inputModel struct {
+	*globalflags.GlobalFlagModel
+	JobName    string
+	InstanceId string
+	Payload    argus.UpdateScrapeConfigPayload
+}
+
+func NewCmd(p *print.Printer) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   fmt.Sprintf("update %s", jobNameArg),
+		Short: "Updates a scrape configuration of an Argus instance",
+		Long: fmt.Sprintf("%s\n%s\n%s",
+			"Updates a scrape configuration of an Argus instance.",
+			"The payload can be provided as a JSON string or a file path prefixed with \"@\".",
+			"See https://docs.api.stackit.cloud/documentation/argus/version/v1#tag/scrape-config/operation/v1_projects_instances_scrapeconfigs_partial_update for information regarding the payload structure.",
+		),
+		Args: args.SingleArg(jobNameArg, nil),
+		Example: examples.Build(
+			examples.NewExample(
+				`Update a scrape configuration from Argus instance "xxx", using an API payload sourced from the file "./payload.json"`,
+				"$ stackit argus scrape-config update my-config --payload @./payload.json --instance-id xxx"),
+			examples.NewExample(
+				`Update an scrape configuration from Argus instance "xxx", using an API payload provided as a JSON string`,
+				`$ stackit argus scrape-config update my-config --payload "{...}" --instance-id xxx`),
+			examples.NewExample(
+				`Generate a payload with the current values of a scrape configuration, and adapt it with custom values for the different configuration options`,
+				`$ stackit argus scrape-config generate-payload --job-name my-config > ./payload.json`,
+				`<Modify payload in file>`,
+				`$ stackit argus scrape-configs update my-config --payload @./payload.json`),
+		),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			ctx := context.Background()
+			model, err := parseInput(cmd, args)
+			if err != nil {
+				return err
+			}
+
+			// Configure API client
+			apiClient, err := client.ConfigureClient(p)
+			if err != nil {
+				return err
+			}
+
+			if !model.AssumeYes {
+				prompt := fmt.Sprintf("Are you sure you want to update scrape configuration %q?", model.JobName)
+				err = p.PromptForConfirmation(prompt)
+				if err != nil {
+					return err
+				}
+			}
+
+			// Call API
+			req := buildRequest(ctx, model, apiClient)
+			_, err = req.Execute()
+			if err != nil {
+				return fmt.Errorf("update scrape config: %w", err)
+			}
+
+			// The API has no status to wait on, so async mode is default
```

Comment:
Does the Portal also indicate the update as done immediatelly?

Reply:
This is something that was also commented on the Terraform provider, but I looked into it now and the endpoint seems to be synchronous. Removed the comment

Reply:
If we know the endpoint is synchronous then we can remove the comment, if it's just fast I think we could keep it

Reply:
The response code is a 202, which would indicate asynchronous. So maybe it's just fast, although the message in the reply states "Successfully updated scrape config".

Reply:
I'll add the comment back